src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1286,12 +1298,23 @@ synchronized public void onChangePredicate(Chat chat, String predicateName, Stri
     for (Session s : sessions.values()) {
       if (s.chat == chat) {
         // found session saving predicates
+        invoke("publishChangePredicate", s, chat, predicateName, result);
         s.savePredicates();
         return;
       }
     }
     error("could not find session to save predicates");
   }
+  
+  public PredicateEvent publishChangePredicate(Session session, Chat chat, String name, String value) {

```

Comment:
why publish the session and the chat...  the chat is a member on the session...  also.. it would be useful to know what the value was prior to it being changed in this call back.  i.e. the original value and the new value..

Reply:
Sure, its just meant to be for convenience to have both.. since PredicateEvent is a POJO its not a big deal to add both
yeah I agree original value should be added too